app/src/main/java/com/checkin/app/checkin/auth/activities/AuthenticationActivity.kt

```diff
@@ -0,0 +1,290 @@
+        flLoadingContainer.visibility = View.VISIBLE
+    }
+
+    private fun hideProgress() {
+        flLoadingContainer.visibility = View.GONE
+    }
+
+    override fun onUserInfoProcess(firstName: String, lastName: String?, username: String, gender: UserModel.GENDER) {
+        authViewModel.register(firstName, lastName, gender, username)
+    }
+
+    override fun onGoogleAuth() {
+        showProgress()
+        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestIdToken(getString(R.string.default_web_client_id))
+                .requestEmail()
+                .requestProfile()
+                .build()
+        val signInIntent = GoogleSignIn.getClient(this, gso).signInIntent
+        startActivityForResult(signInIntent, RC_AUTH_GOOGLE)
+    }
+
+    override fun onFacebookAuth(loginResult: LoginResult) {
+        showProgress()
+        val accessToken = loginResult.accessToken
+        authViewModel.setProviderIdToken(accessToken.token)
+        val credential = FacebookAuthProvider.getCredential(accessToken.token)
+        authenticateWithCredential(credential)
+    }
+
+    override fun onPhoneAuth(phoneNo: String) {
+        val action = AuthOptionFragmentDirections.actionVerifyOtp(phoneNo)
+        navController.navigate(action)
+    }
+
+    private fun successAuth(data: AuthResultModel) {
+        Handler().post {
+            val authToken = data.token
+            val account = Account(resources.getString(R.string.app_name), Constants.ACCOUNT_TYPE)
+            val accountManager = AccountManager.get(this)
+            val userData = Bundle()
+            userData.putLong(Constants.ACCOUNT_UID, data.userId)
+            accountManager.addAccountExplicitly(account, null, userData)
+            accountManager.setAuthToken(account, AccountManager.KEY_AUTHTOKEN, authToken)
+            startService(Intent(applicationContext, DeviceTokenService::class.java))
+            userViewModel.userData.observe(this, Observer {
+                it?.let {
+                    when (it.status) {
+                        Resource.Status.SUCCESS -> {
+                            if (it.data?.firstName == defaultFirstName) {
+                                fetchUserDetails(name = null)
+                                //so if the firebase has  a name then that name would be the default name and the app will next time open with the
+                                //name firebase gave otherwise if the username is Checkin(default name) then the case would be that the name would be checked
+                                //if so that the name is checkin then the details screen would be fetched
+                            } else {
+                                openHome()
+                            }
+                        }
+                    }
+                }
+            })
+            userViewModel.fetchUserData()
+        }
+    }
+
+    private fun openHome() {
```

Comment:
Instead of creating a new function, reuse an [existing one](https://github.com/checkin-team/Checkin-Android/blob/3454cf0730aeedcceecadae5e66c0133175e323e/app/src/main/java/com/checkin/app/checkin/utility/Utils.java#L117)